packages/ui-causal-graph-editor/src/shared/graph-layout/planar-layout.tsx

```diff
@@ -74,30 +113,50 @@ export default class PlanarLayout extends GraphLayout {
             newLayout: LayoutMapping<XYPosition>;
             onAddNode?: () => void | Promise<void>;
         } => {
-            const dag = dagGraphParser(currentGraph);
+            const dag = dagGraphParser(currentGraph, this.tiers);
 
             /**
              * The nodeSize is scaled for consistent spacing in the horizontal layout
              */
-            const newDagLayout = sugiyama<DagNode<DagNodeData>>()
-                .nodeSize(() => [this.nodeSize * 3, this.nodeSize * 6])
-                .coord(coordQuad() as any)(dag);
-
-            const edgePoints = newDagLayout.dag.links().reduce((acc, link) => {
-                acc[`${link.source.id}||${link.target.id}`] = link.points.map((point) => ({
-                    x: this.orientation === 'vertical' ? point.x : point.y,
-                    y: this.orientation === 'vertical' ? point.y : point.x,
+            let newDagLayout;
+
+            try {
+                function groupAccessor(node: any): string {
+                    return node.data.group;
+                }
+
+                function rankAccessor(node: any): number {
+                    return node.data.rank;
+                }
+
+                newDagLayout = sugiyama()
+                    .nodeSize(() => [this.nodeSize * 3, this.nodeSize * 6])
+                    .coord(coordQuad() as any)
+                    .layering(
+                        this.tiers ? layeringSimplex().group(groupAccessor).rank(rankAccessor) : layeringSimplex()
+                    )
+                    .decross(this.tiers ? customDecross : decrossTwoLayer());
+
+                newDagLayout(dag);
+            } catch (e) {
+                throw new Error('d3-dag failed to resolve the layering of graph nodes for PlanarLayout.');
+            }
+
+            const edgePoints: LayoutMapping<XYPosition[]> = {};
+            for (const link of dag.links()) {
+                edgePoints[`${link.source.data.id}||${link.target.data.id}`] = link.points.map((point: number[]) => ({
+                    x: this.orientation === 'vertical' ? point[0] : point[1],
+                    y: this.orientation === 'vertical' ? point[1] : point[0],
                 }));
-                return acc;
-            }, {} as LayoutMapping<XYPosition[]>);
+            }
 
-            const newLayout = newDagLayout.dag.descendants().reduce((acc, node) => {
-                acc[node.id] = {
+            const newLayout: LayoutMapping<XYPosition> = {};
+            for (const node of dag.nodes()) {
+                newLayout[node.data.id] = {
                     x: this.orientation === 'vertical' ? node.x : node.y,
                     y: this.orientation === 'vertical' ? node.y : node.x,
                 };
-                return acc;
-            }, {} as LayoutMapping<XYPosition>);
+            }
```

Comment:
It was because dag.nodes() is iterable so I couldn't apply reduce to it directly. Have updated to convert to an array first so we can still use reduce.